src/tools/fuzzing/fuzzing.cpp

```diff
@@ -1688,6 +1691,73 @@ Expression* TranslateToFuzzReader::makeTry(Type type) {
   return builder.makeTry(body, catchTags, catchBodies);
 }
 
+Expression* TranslateToFuzzReader::makeTryTable(Type type) {
+  if (funcContext->breakableStack.empty()) {
+    // Nothing to break to.
+    // TODO: Perhaps generate a block wrapping us?
+    return makeTrivial(type);
+  }
+  if (wasm.tags.empty()) {
+    addTag();
+  }
+
+  // Add catches of specific tags, and possibly a catch_all at the end. We use
+  // the last iteration of the loop for that.
+  std::vector<Name> catchTags;
+  std::vector<Name> catchDests;
+  std::vector<bool> catchRefs;
+  auto numTags = upTo(MAX_TRY_CATCHES);
+  for (Index i = 0; i <= numTags; i++) {
+    Name tagName;
+    Type tagType;
+    if (i < numTags) {
+      // Look for a specific tag.
+      auto& tag = pick(wasm.tags);
+      tagName = tag->name;
+      tagType = tag->sig.params;
+    } else {
+      // Look for a catch_all at the end, some of the time (but all of the time
+      // if we have nothing else).
+      if (!catchTags.empty() && oneIn(2)) {
+        break;
+      }
+      tagType = Type::none;
+    }
+
+    // We need to find a proper target to break to, which means a target that
+    // has the type of the tag, or the tag + an exnref at the end.
+    std::vector<Type> vec;
+    for (auto t : tagType) {
+      vec.push_back(t);
+    }
+    vec.push_back(Type(HeapType::exn, Nullable)); // TODO: NonNullable?
```

Comment:
Isn't `exnref` `ref null exn`? Why would we need a nonnullable exnref?

Reply:
I wasn't sure what the spec allows. Don't we always send a non-nullable exnref, if an exception is thrown? If so then the target block could be non-nullable too, like this?
```wat
(block $block (ref exn) ;; non-nullable
  (try_table (catch_all $block)
    ..
  )
)
```
Or does the spec say to use a nullable value for some reason?

Reply:
I'm not sure. Given that exnref values generated from `try_table` will never be null, I think allowing the non-nullable type makes sense though. I asked the question in the EH repo: https://github.com/WebAssembly/exception-handling/issues/336

Reply:
It looks the non-nullable version is allowed:
https://github.com/WebAssembly/exception-handling/issues/336#issuecomment-2398019414

Reply:
Oh cool, thanks. I added support for that now.